src/iterative_and_periodic.jl

```diff
@@ -177,8 +182,7 @@ end
     # Checking for floating point equality is OK here as `DifferentialEquations.jl`
     # sets the time exactly to the final time in the last iteration
     return integrator.t == last(integrator.sol.prob.tspan) ||
-           isempty(integrator.opts.tstops) ||
```

Comment:
@ChrisRackauckas is this really needed? As far as I can tell `integrator.opts.tstops` is never empty for ODEs as it always contains `tspan[2]` too, whereas in SSAStepper we do not insert `tspan[2]` into it (we actually don't copy the user `tstops` over, which we probably should do in cases that we insert into it but that is another issue).

If I remove this check everything seems to work ok in my limited testing for both jumps and ODEs (with the JumpProcesses PR I'll open).

Reply:
i.e. https://github.com/SciML/OrdinaryDiffEq.jl/blob/dc0e1e733fdd79a6c643cbd0c4f4be03c934cb91/lib/OrdinaryDiffEqCore/src/solve.jl#L609 adds `tspan[2]` into the internal ODE `tstops` which we never do for SSAStepper.

Reply:
There is a case where tstops is empty which is when a user manually takes control over stepping in the integrator interface. You can then have an `Inf` or `nothing` tspan. This is used for example in deployments of control systems, where the integrator is effectively set to just run forever based on a fixed timer for the control loop, reading in data and sending out a model-predicted reaction. This is somewhat different from the normal open source simulation usage but these deployments are something that's becoming a lot more common.

That said, this condition would then force the callback to end in that scenario, which seems like it's counter-productive to wanting that to never finish. So from what I can see, I think it's safe to remove this condition.